CONTRIBUTING.md

```diff
@@ -44,6 +44,8 @@ Keep these in mind as both authors and reviewers of PRs:
   - The PR must contain related changes and leave out unrelated changes (e.g. refactorings, etc. that could be their own PR instead).
   - If the PR could be broken up into two or more PRs either "vertically" (by separating concerns logically) or horizontally (by sharding the PR into a series of PRs --- usually works well with mass refactoring or cleanup type PRs), it should. A set of such related PRs can be tracked and given context in a meta issue.
 
+* The PR changeset should be rather small: 500 lines of code (excluding generated code or other changes that normally should not require close review by a human) is the soft limit that should be used to judge if a PR is "too big".
```

Comment:
LGTM :+1: 500 lines is also used in [OTel Collector guidelines](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.114.0/CONTRIBUTING.md#how-to-structure-prs-to-get-expedient-reviews):

> We recommend that any PR (unless it is trivial) to be smaller than 500 lines (excluding go mod/sum changes) in order to help reviewers to do a thorough and reasonably fast reviews.